lib/rubocop/cop/rails/has_many_or_has_one_dependent.rb

```diff
@@ -41,11 +41,11 @@ class HasManyOrHasOneDependent < Base
         PATTERN
 
         def_node_matcher :association_without_options?, <<~PATTERN
-          (send nil? {:has_many :has_one} _)
+          (send _ {:has_many :has_one} _)
         PATTERN
 
         def_node_matcher :association_with_options?, <<~PATTERN
-          (send nil? {:has_many :has_one} ... (hash $...))
+          (send _ {:has_many :has_one} ... (hash $...))
```

Comment:
I'm concerned that accepting any receiver would increase false positives potential, but common user would refrain from creating `has_many` and `has_one` business methods in Rails. So Let's try this first.